### PR TITLE
 Feature/api search by people #2669 

### PR DESCRIPTION
**Pull request by AWare**
this is #2669
also includes all the commits from https://github.com/guardian/grid/pull/2956

# Original PR
This requires both https://github.com/guardian/grid/pull/2664 and https://github.com/guardian/grid/pull/2667 to be merged first.

## What does this change?
Allows for you to search for a 'person' using either the api or via the UI. It also displays the people in an image on the ui in the metadata section if there are any people present. When searching for the name of a person it must be an exact match and case sensitive - probably something we would want to improve on later on.

Screenshot 2019-10-25 at 12 16 10

Screenshot 2019-10-25 at 12 16 19

### CODE REVIEW

**Comment by AWare** [comment]
The original metadata change failed on prod as `getMetadataDiff` could not handle arrays, and special cased the known arrays. This adds a test suite for that function, which should handle all the functionality that's described in it, and then replaces the implementation with something simpler which should handle arrays in any position.

File: kahuna/public/js/edits/metadataDiff.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    Object.keys(metadata).concat(Object.keys(originalMetadata))
+  );
+
+  keys.forEach((key) => {
+    if (originalMetadata[key] === metadata[key]) {
+      return;
+    }
+
+    //This only works with string fields and does not support arrays
+    if (Array.isArray(metadata[key]) || Array.isArray()) {
```
Comment (jonathonherbert) [inline kahuna/public/js/edits/metadataDiff.js:15]: `Array.isArray()` will always return false – what's the purpose of this second condition?

Comment (AWare) [reply kahuna/public/js/edits/metadataDiff.js]: That's supposed to be checking the other potential value for an array

File: kahuna/public/js/edits/metadataDiff.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+export const getMetadataDiff = (image, metadata) => {
+  const diff = {};
+  const originalMetadata = image.data.originalMetadata;
+  // jscs has a maximumLineLength of 100 characters, hence the line break
+  var keys = new Set(
+    Object.keys(metadata).concat(Object.keys(originalMetadata))
+  );
+
+  keys.forEach((key) => {
+    if (originalMetadata[key] === metadata[key]) {
```
Comment (jonathonherbert) [inline kahuna/public/js/edits/metadataDiff.js:10]: This logic is expressed much more cleanly here – as far as I can see it's like for like 🤞

**Pull request merged**